include/re_sipsess.h

```diff
@@ -7,6 +7,15 @@
 struct sipsess_sock;
 struct sipsess;
 
+/* SDP Negotiation state */
+enum sdp_neg_state {
+	SDP_NEG_NONE = 0,
+	SDP_NEG_LOCAL_OFFER = 1,	/** SDP offer sent */
+	SDP_NEG_REMOTE_OFFER = 2,	/** SDP offer received */
+	SDP_NEG_PREVIEW_ANSWER = 6,	/** SDP preview answer sent */
+	SDP_NEG_DONE = 8		/** SDP negotiation done */
+};
```

Comment:
Can you explain why there are gaps in the enumerations? Are some missing? Or can be combined? Like flags?

Reply:
A nicer technique for independent flags is this:

```c
enum sdp_neg_state {
	SDP_NEG_NONE = 0,
	SDP_NEG_LOCAL_OFFER = (1 << 0),	/** SDP offer sent */
	SDP_NEG_REMOTE_OFFER = (1 << 1),	/** SDP offer received */
	SDP_NEG_PREVIEW_ANSWER = (1 << 2),	/** SDP preview answer sent */
	SDP_NEG_DONE = (1 << 3)		/** SDP negotiation done */
};
```